base_name_search_improved/models/ir_model.py

```diff
@@ -20,35 +20,38 @@ def make_name_search():
             @api.model
             def name_search(self, name='', args=None,
                             operator='ilike', limit=100):
+                # Reusable parts
+                def extend_name_results(domain, results, limit):
+                    if len(results) < limit:
+                        domain += [('id', 'not in', [x[0] for x in results])]
+                        recs = self.search(domain, limit=limit - len(results))
```

Comment:
If you store `len(results)` in a variable, you get another ridiculous performance boost avoiding to count them twice.